onadata/apps/logger/migrations/0015_add_delete_data_permission.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.contrib.auth.management import create_permissions
+from django.contrib.auth.models import (
+    Permission,
+    User,
+    AnonymousUser,
+)
+
+
+def forwards_func(apps, schema_editor):
+    """
+    All users need to receive the new permission at the model level.
+    """
+    # Permission does not exist when running this migration for the first time.
+    # Django is running migrations in a transaction and permissions are created
+    # after the transaction is completed.
+
+    # ToDo update this code when upgrading to Django 2.x
+    # see https://stackoverflow.com/a/40092780/1141214
+    apps.models_module = True
+    create_permissions(apps, verbosity=0)
+    apps.models_module = None
+
+    permission = Permission.objects.get(codename='delete_data_xform')
+    users = User.objects.exclude(pk=AnonymousUser().pk)
+    for user_ in users.all():
+        user_.user_permissions.add(permission)
+
+
+def reverse_func(apps, schema_editor):
+    users = User.objects.exclude(pk=AnonymousUser().pk)
+    permission = Permission.objects.get(codename='delete_data_xform')
+    for user_ in users.all():
+        user_.user_permissions.remove(permission)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('logger', '0014_attachment_add_media_file_size'),
+    ]
+
+    operations = [
+        migrations.AlterModelOptions(
+            name='xform',
+            options={'ordering': ('id_string',), 'verbose_name': 'XForm', 'verbose_name_plural': 'XForms', 'permissions': (('view_xform', 'Can view associated data'), ('report_xform', 'Can make submissions to the form'), ('move_xform', 'Can move form between projects'), ('transfer_xform', 'Can transfer form ownership'), ('validate_xform', 'Can validate submissions'), ('delete_data_xform', 'Can delete submissions'))},
+        ),
+        # This can take a while on big databases
```

Comment:
I wonder if `bulk_create()` would be an option (and if it would save time)
https://docs.djangoproject.com/en/1.8/ref/models/querysets/#django.db.models.query.QuerySet.bulk_create

We'd have to get at the "through" model, `User.user_permissions.through`: https://stackoverflow.com/a/10116452/2402324

Reply:
It took ~3m30 on `kf.beta`. Let's assume it could take up to ~7m on our production servers. 

With your `bulk_create()` suggestion, it took ~11s on `kf.beta`. Much better ;-)

Reply:
It took 3 minutes on `kf.beta`. I think it could take ~ twice the time on OCHA server. Slow but not that bad. 

With `bulk_create` and `through` model: It took 11s.